ietfYangDraftPull/pull_local.py

```diff
@@ -107,63 +110,32 @@ def main(script_conf: ScriptConfig = DEFAULT_SCRIPT_CONFIG.copy()) -> list[JobLo
+    yang_models_dir = config.get('Directory-Section', 'yang-models-dir')
+    non_ietf_directory = config.get('Directory-Section', 'non-ietf-directory')
+    openconfig_dir = os.path.join(non_ietf_directory, 'openconfig/public')
     logger = log.get_logger('pull_local', f'{log_directory}/jobs/pull-local.log')
     logger.info('Starting cron job IETF pull request local')
 
     messages = []
     notify_indexing = notify_indexing == 'True'
     success = True
-    try:
-        # Clone YangModels/yang repository
-        clone_dir = os.path.join(temp_dir, 'pull_local')
-        if os.path.exists(clone_dir):
-            shutil.rmtree(clone_dir)
-        repo = repoutil.ModifiableRepoUtil(
-            f'{github_url}/YangModels/yang.git',
-            clone_options=repoutil.RepoUtil.CloneOptions(
-                config_username=config_name,
-                config_user_email=config_email,
-                local_dir=clone_dir,
-            ),
-        )
-        logger.info(f'YangModels/yang repo cloned to local directory {repo.local_dir}')
-
-        response = requests.get(ietf_rfc_url)
-        tgz_path = os.path.join(repo.local_dir, 'rfc.tgz')
-        extract_to = os.path.join(repo.local_dir, 'standard/ietf/RFC')
-        with open(tgz_path, 'wb') as zfile:
-            zfile.write(response.content)
-        tar_opened = draftPullUtility.extract_rfc_tgz(tgz_path, extract_to, logger)
-
-        if tar_opened:
-            # Standard RFC modules
-            directory_success, message = populate_directory(extract_to, notify_indexing, logger)
-            success = success and directory_success
-            messages.append({'label': 'Standard RFC modules', 'message': message})
-
-        # Experimental modules
-        experimental_path = os.path.join(repo.local_dir, 'experimental/ietf-extracted-YANG-modules')
-
-        directory_success, message = populate_directory(experimental_path, notify_indexing, logger)
-        success = success and directory_success
-        messages.append({'label': 'Experimental modules', 'message': message})
-
-        # IANA modules
-        iana_path = os.path.join(repo.local_dir, 'standard/iana')
-
-        if os.path.exists(iana_path):
-            directory_success, message = populate_directory(iana_path, notify_indexing, logger)
-            success = success and directory_success
-            messages.append({'label': 'IANA modules', 'message': message})
-
-    except Exception as e:
-        logger.exception('Exception found while running pull_local script')
-        raise e
+
+    with WorktreeManager() as worktrees:
+        try:
+            for repo, path_in_repo, label in (
+                (yang_models_dir, 'standard/ietf/RFC', 'Standard RFC modules'),
+                (yang_models_dir, 'experimental/ietf-extracted-YANG-modules', 'Experimental modules'),
+                (yang_models_dir, 'standard/iana', 'IANA modules'),
+                (openconfig_dir, '', 'OpenConfig modules'),
```

Comment:
what about specific logic for populating openconfig modules, is it just removed?